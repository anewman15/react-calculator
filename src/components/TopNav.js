/*
 eslint-disable no-unused-vars
*/

import React from 'react';
import { Navbar, Nav } from 'react-bootstrap';

const TopNav = () => (
  <div>
    <Navbar bg="success" expand="lg">
      <Navbar.Brand className="text-white font-weight-bolder" href="#home">No Magic - Dumb Calculator</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="ml-auto text-white">
          <Nav.Link className="text-white" href="#home">Home</Nav.Link>
          <Nav.Link className="text-white" href="#link">Calculator</Nav.Link>
          <Nav.Link className="text-white" href="#link">Quote of the Day</Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  </div>
);

export default TopNav;