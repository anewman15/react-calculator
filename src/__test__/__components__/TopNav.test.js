/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import TopNav from '../../components/TopNav';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('TopNav', () => {
  it('should render a navigation bar at the top of the site', () => {
    const tree = renderer
      .create(<TopNav />)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});