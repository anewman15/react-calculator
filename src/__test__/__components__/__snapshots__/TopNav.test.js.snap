// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav should render a navigation bar at the top of the site 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-success"
  >
    <a
      className="text-white font-weight-bolder navbar-brand"
      href="/"
    >
      Over the Counter Calculator
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ml-auto text-white navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="text-white nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="text-white nav-link"
          data-rb-event-key="/calculator"
          href="/calculator"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calculator
        </a>
        <a
          className="text-white nav-link"
          data-rb-event-key="/quote-of-the-day"
          href="/quote-of-the-day"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Quote of the Day
        </a>
      </div>
    </div>
  </nav>
</div>
`;
